server.js

```diff
@@ -17,16 +13,92 @@ mongoose.Promise = Promise;
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
-const port = process.env.PORT || 8080;
+const port = process.env.PORT || 8081;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Req: Your API should make use of Mongoose models to model your data and use these models to fetch data from the database.
+// Modelling the database:
+const Laureate = mongoose.model("Laureate", {
+  id: Number,
+  name: String,
+  year: Number,
+  country: String,
+  category: String,
+  description: String,
+});
+
+// Codealong - add to database code:
+// const testUser = new User({
+//   name: "Justyna Zwiazek",
+//   id: 65,
+//   category: "Peace",
+// });
+// console.log(testUser);
+// testUser.save();
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    // deleti all items in the database to prevent to copy over the items
+    await Laureate.deleteMany();
+
+    femalesData.forEach((female) => {
+      const newLaureate = new Laureate(female);
+      newLaureate.save();
+    });
+  };
+  seedDatabase();
+}
+// Endpoints on the landing page
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+// Req: A minimum of one endpoint to return a collection of results (array of elements)
+app.get("/laureates", async (req, res) => {
+  const { name, category } = req.query;
+
+  try {
+    const allLaureates = await Laureate.find({
+      name: new RegExp(name, "i"),
+      category: new RegExp(category, "i"),
+    });
+    res.json(allLaureates);
+  } catch (error) {
+    res
+      .status(400)
+      .json({ error: "We didn't find what you were looking for." });
+  }
+});
+
+// Req: A minimum of one endpoint to return a single result (single element).
+app.get("/laureates/:id", async (req, res) => {
```

Comment:
This endpoint only works for id: 1 unfortunately, when searching for id 9 or 24 for example, nr 1 still shows up!